pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -795,6 +795,15 @@ allChainsLabel:
 			filterChainMatch = &fcm
 			if filterChainMatchOption[id].Protocol == istionetworking.ListenerProtocolHTTP {
 				httpOpts = configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, pluginParams)
+				if chain.TLSContext != nil && chain.TLSContext.CommonTlsContext != nil {
+					alpnProtocols := chain.TLSContext.CommonTlsContext.AlpnProtocols
+					for i, alpn := range alpnProtocols {
+						if alpn == "istio-peer-exchange" {
+							copy(alpnProtocols[i:], alpnProtocols[i+1:])
```

Comment:
what if i is the last element in the array? wont i+1 throw out of bounds?

Reply:
Thanks..Fixed